tests/test_Summary.cpp

```diff
@@ -1063,6 +1063,51 @@ BOOST_AUTO_TEST_CASE(fpr) {
     BOOST_CHECK_CLOSE( fpr2_si, ecl_sum_get_field_var( resp2, 1, "FPR" ), 1e-5 ); //
 }
 
+BOOST_AUTO_TEST_CASE(fprp) {
+    setup cfg( "test_fprp", "summary_deck_non_constant_porosity.DATA");
+
+    out::Summary writer( cfg.es, cfg.config, cfg.grid, cfg.schedule, cfg.name );
+    writer.add_timestep( 0, 0 * day, cfg.es, cfg.schedule, cfg.wells , cfg.solution, {});
+    writer.add_timestep( 1, 1 * day, cfg.es, cfg.schedule, cfg.wells , cfg.solution, {});
+    writer.add_timestep( 2, 2 * day, cfg.es, cfg.schedule, cfg.wells , cfg.solution, {});
+    writer.write();
+
+    auto res = readsum( cfg.name );
+    const auto* resp = res.get();
+
+    UnitSystem units( UnitSystem::UnitType::UNIT_TYPE_METRIC );
+    // fprp = sum_ (p * pv ) / pv
+    const double fprp_si =  ( (3 * 0.1 + 8 * 0.2) * 500 ) / ( (500*0.1 + 500*0.2));
+    const double fprp = units.from_si( UnitSystem::measure::pressure, fprp_si );
+    BOOST_CHECK_CLOSE( fprp, ecl_sum_get_field_var( resp, 1, "FPRP" ), 1e-5 ); //
+
+    // change pressure and test again.
+    size_t g = 0;
+    for (size_t k=0; k < cfg.grid.getNZ(); k++) {
```

Comment:
This is of course perfectly OK - but to to be slightly anal I would suggest looping over the active index instead:
```C++
for (size_t g=0; g < grid.getNumActive(); g++)
     cfg.solution.data("PRESSURE")[g] = units.to_si( ... )
```
in the input and output handling there is certainly some use of the cartesian indices, but in the simulator as such there is really no concept of a cartesian coordinate system and all sums are only over active cells.